src/HealthChecks.OpenIdConnectServer/IdSvrHealthCheck.cs

```diff
@@ -12,16 +12,22 @@ public class IdSvrHealthCheck
         const string IDSVR_DISCOVER_CONFIGURATION_SEGMENT = ".well-known/openid-configuration";
 
         private readonly Func<HttpClient> _httpClientFactory;
-        public IdSvrHealthCheck(Func<HttpClient> httpClientFactory)
+        private readonly string _prefix;
+        public IdSvrHealthCheck(Func<HttpClient> httpClientFactory, string prefix)
         {
             _httpClientFactory = httpClientFactory ?? throw new ArgumentNullException(nameof(httpClientFactory));
+            _prefix = prefix;
         }
         public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
         {
             try
             {
                 var httpClient = _httpClientFactory();
-                var response = await httpClient.GetAsync(IDSVR_DISCOVER_CONFIGURATION_SEGMENT, cancellationToken);
+                var requestUri = string.IsNullOrEmpty(_prefix)
+                    ? IDSVR_DISCOVER_CONFIGURATION_SEGMENT
+                    : $"{_prefix}/{IDSVR_DISCOVER_CONFIGURATION_SEGMENT}";
```

Comment:
Maybe just move `IDSVR_DISCOVER_CONFIGURATION_SEGMENT` from constant into new parameter and use as default value?

Reply:
Also note that the same configuration may be done via provided `httpClientFactory` so this PR not needed.

Reply:
`AddHttpClient` calls are cumulative so you can allready configure your http client to use any prefix you want.

Reply:
I changed the prefix parameter into the requestUri parameter, but anyone that overrides it should also include .well-known/openid-configuration in the requestUri value now.
Also: how would this work without this PR? It's not enough to set the BaseAddress of the HttpClient. Can I set the request uri of `GetAsync` in `httpClientFactory`?

Reply:
If this works for your case then I really think this PR not needed.

Reply:
Before I made the PR, I did a test in a console app and thought that it didn't work with only setting BaseAddress.
It turns out that I had to end the BaseAddress with a slash which I didn't think of initially.
So when I set the base address to `https://dev.k8s-local.my-app.com/identityserver**/**', it actually works.
Looks like you're right that this PR is not needed. Shall I close it?

Reply:
Yep.